frontend/src/modules/WellLogViewer/settings/atoms/derivedAtoms.ts

```diff
@@ -0,0 +1,107 @@
+import { WellboreHeader_api, WellboreLogCurveHeader_api } from "@api";
+import { EnsembleSetAtom } from "@framework/GlobalAtoms";
+import { WellPicksLayerData } from "@modules/Intersection/utils/layers/WellpicksLayer";
+import { TemplatePlot, TemplateTrack } from "@webviz/well-log-viewer/dist/components/WellLogTemplateTypes";
+
+import { atom } from "jotai";
+import _, { Dictionary } from "lodash";
+
+import {
+    userSelectedFieldIdentifierAtom,
+    userSelectedNonUnitWellpicksAtom,
+    userSelectedUnitWellpicksAtom,
+    userSelectedWellboreUuidAtom,
+} from "./baseAtoms";
+import { logViewerTrackConfigs } from "./persistedAtoms";
+import {
+    drilledWellboreHeadersQueryAtom,
+    wellLogCurveHeadersQueryAtom,
+    wellborePicksAndStratigraphyQueryAtom,
+} from "./queryAtoms";
+
+export const selectedEnsembleSetAtom = atom((get) => {
```

Comment:
Filtered doesn't seem quite right either. What's wrong is the use of "set" when I'm actually picking one from the set, so I'll rename it to `selectedEnsembleAtom`

Reply:
My mistake, I assumed you selected all ensembles for the given field identifier - misread the `find` for a `filter` due to expectation. That's also why my second comment is not quite right. However, you are always using the first ensemble for the selected field identifier. What if you are having multiple ensembles for the same field? You are using the ensemble's case uuid to fetch stratigraphy. Apparently, we cannot assume that stratigraphy is constant among all cases (why would it otherwise be a parameter in the request). Should it be possible for the user to switch the case?